core/test/utils.test.js

```diff
@@ -53,6 +53,12 @@ suite('utils', () => {
         createNestedObject(obj, ['first', 'second', 'third', 'fourth']);
         expect(obj).to.deep.equal({ first: { second: { third: { fourth: {} } } } });
       });
+
+      test('Should create an object with not object fields', () => {
```

Comment:
Guess I'm missing something here. 😅 
So, if I understood your scenario correctly, should we adapt the code so at the end we have an empty object on `prop1`?  Something like: `{ prop1: {}, prop2: 'value'}`

Reply:
I guess what this tests is that if you input an object and a path to a property which is already initialized, it basically just returns a duplicate of the input object. While the test is valid (even if the description needs to be rephrased), I'm not 100% sure that we consider a path to a property that is already initialized as valid input to a method whose goal is to initialize that property... Any thoughts?